### PR TITLE
Simple dialogue priority implementation

### PR DESCRIPTION
**Pull request by eftg1000**
# Description

The dialogue prioritisation has been implemented where the item pick up dialogue does not override the conversation with the NPC dialogue.

Also the method override for the updateText(String[]) method has been deleted since it is now unnecessary.

Fixes / Closes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Simple priority implementation for the dialogues.
- [x] Deletion of the method updateText which takes an array of String.

# How Has This Been Tested?

- [x] Tried walking towards an item while the NPC conversation is happening and also the vice versa.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/services/DialogueBoxService.java [hunk 1]
```diff
@@ -56,11 +58,32 @@ public Boolean getIsVisible() {
         return currentOverlay.getIsVisible();
     }
 
+    /**
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/services/DialogueBoxService.java:61]: This seems like an extremely overcomplicated way to do this - is it really necessary to check the call stack? Can you not just create an enum in here which map to priority values (ie call updateText(String[] text, DialoguePriorityEnum))? The current changes seems to increase coupling - if anyone changes the name of any of these classes they'll need to remember to change it here as well.

**Review by mitsudz — CHANGES_REQUESTED** [review]
See associated comment - this seems like an overcomplicated and potentially dangerous way to implement an extremely simple feature imo.

File: source/core/src/main/com/csse3200/game/services/DialogueBoxService.java (hunk 1 above)
Comment (eftg1000) [reply source/core/src/main/com/csse3200/game/services/DialogueBoxService.java:61]: By DialoguePriorityEnum, you mean like an integer value right?

Comment (mitsudz) [reply source/core/src/main/com/csse3200/game/services/DialogueBoxService.java:61]: Yes

**Review by LucasRyall — COMMENTED** [review]
I agree with Mittun, each dialogue box type should have a priority, with higher priority ones overriding lower priority ones, with his suggestion of a DialoguePriorityEnum souning like a good solution. another solution (given that you've only got 2 different types of dialogue boxes) is to pass updateText a boolean, false doesn't override current text (being called by item), and true does override current text (being called by npc), this only works if only 2 things are using the dialogue box service.

**Review by mitsudz — CHANGES_REQUESTED** [review]
This looks much better, but can you create an enum in the DialogueBoxService which the other classes can access, or at least give names to the priority values rather than having magic numbers everywhere?

File: source/core/src/main/com/csse3200/game/services/DialogueBoxService.java (hunk 1 above)
Comment (mitsudz) [reply source/core/src/main/com/csse3200/game/services/DialogueBoxService.java]: Have re-reviewed - only one small change requested now.

**Review by mitsudz — APPROVED** [review]

**Review by LucasRyall — APPROVED** [review]
Changes look good.

**Pull request merged**